api/src/main/java/org/commonjava/indy/data/StoreValidatorRemote.java

```diff
@@ -0,0 +1,162 @@
+                // Waiting for Http GET & HEAD Request Executor tasks to finish
+                httpRequestsLatch.await();
+                // Check for Sucessfull Validation
+                if (httpGetStatus.get() && httpHeadStatus.get()) {
+                    LOGGER.info("=> Succesfull Validation for Remote Repository: " + remoteUrl.get());
+                    return new ArtifactStoreValidateData
+                        .Builder(true)
+                        .setRepositoryUrl(remoteUrl.get().toExternalForm())
+                        .setStoreType(remoteRepository.getType())
+                        .build();
+                }
+            } else {
+                LOGGER.warn("=> It's not Remote Artifact Store Repository: " + artifactStore.getType().singularEndpointName());
+            }
+            //#endregion
+        } catch (MalformedURLException mue) {
+            LOGGER.error("=> Mailformed URL: ", mue);
+            throw new MalformedURLException("=> Mailformed URL: " + ((RemoteRepository) artifactStore).getUrl());
+        } catch (Exception e) {
+            LOGGER.error(" => Not Valid Remote Repository, \n => Exception: " + e);
+            throw new InvalidArtifactStoreException("=> Not valid remote Repository: " + artifactStore.getType().singularEndpointName(), e);
+        } finally {
+            if (executorService.isShutdown()) {
+                try {
+                    executorService.awaitTermination(60, TimeUnit.SECONDS);
+                } catch (InterruptedException ie) {
+                    LOGGER.warn("=> Some of Http Executors Tasks are still in waiting state after 60s timeout");
+                    executorService.shutdownNow();
+                }
+            }
+        }
+
+        if (artifactStore instanceof RemoteRepository) {
+            return new ArtifactStoreValidateData
+                .Builder(false)
+                .setRepositoryUrl(((RemoteRepository) artifactStore).getUrl())
+                .setStoreType(artifactStore.getType())
+                .build();
+        } else if (artifactStore instanceof HostedRepository) {
+            return new ArtifactStoreValidateData
+                .Builder(false)
+                .setRepositoryUrl(((HostedRepository) artifactStore).getStorage())
+                .setStoreType(artifactStore.getType())
+                .build();
+        } else {
+            return new ArtifactStoreValidateData
+                .Builder(false)
+                .setRepositoryUrl("/")
+                .setStoreType(artifactStore.getType())
+                .build();
+        }
+
+    }
+
+    private boolean allowedGet(List<String> allowed) {
+        return allowed.indexOf("GET") != -1 ? true : false;
+    }
+
+    private boolean allowedHead(List<String> allowed) {
+        return allowed.indexOf("HEAD") != -1 ? true : false;
+    }
+
+    class HttpGetTask implements Callable<Boolean> {
```

Comment:
If the only difference between this Callable and the HEAD Callable is the method added to the logging lines, can we extract the method from the HttpRequest object, or else pass it in as a constructor parameter, and reduce this to a single generalized class?